internal/plan/plan.go

```diff
@@ -26,18 +26,33 @@ import (
 	"strings"
 	"time"
 
+	"github.com/imdario/mergo"
+	"github.com/mitchellh/copystructure"
```

Comment:
In our conversation I asked whether we could just use reflection, and I thought the positive reply meant the internal reflect package. I didn't realize that by "reflection" you were actually referring to embedding two other packages which introduce plenty of complexity of their own.  Can we please drop these dependencies for this PR, and split out that conversation out?

Reply:
Ah, I hadn't realized that. I did refer to the two reflect-based libraries in the meeting, but I understand the confusion. I've reverted to the hand-rolled logic for now and we can discuss further in a follow-up PR later.